daemons/based/based_notify.c

```diff
@@ -200,8 +188,6 @@ cib_diff_notify(const char *op, int result, const char *call_id,
     crm_xml_add(update_msg, PCMK__XA_SRC, origin);
     crm_xml_add_int(update_msg, PCMK__XA_CIB_RC, result);
 
-    attach_cib_generation(update_msg, "cib_generation", the_cib);
-
```

Comment:
CIB notification callbacks have access to this information, even if we're not using it internally at this point, so it should be kept for external compatibility. I don't think we've ever documented what callbacks get in the XML, but just to be extra safe.

Ditto with all the other stuff dropped here

Reply:
> I don't think we've ever documented what callbacks get in the XML

which was why I felt comfortable with this...

There are probably zero external callers; if there are any and they're looking for specific fields, then they're prone to breakage because we never promised specific fields; and there's no way to deprecate these fields because we never documented them in the first place.

sigh, how do I get rid of something we never told anyone about... create a schema (somewhere, I guess in `xml/api`) for XML notifications for the express purpose of deprecating part of it?

Reply:
Nope, just hold it for 3.0.0 :)

Reply:
That's simple enough. Still, it sort of kicks the can down the road. Assuming we haven't deprecated the `cib_t` public API by 3.0.0, we'll be in the same position in 3.0.0 if we ever want to remove unused and undocumented data fields (unlikely as it may be).

At least this will get rid of the message data block containing the update request. Copying all that XML for every notification presumably eats up a non-trivial amount of resources.

Reply:
Let me know if you have any thoughts re: kicking the can, like whether the notification content should be documented somewhere or schema-fied, or try to deprecate `cib_t` as public ASAP... even though no one is likely using it, I'd be surprised if we did that until libpacemaker is more complete.

Reply:
We can leave it undocumented